systemd: Change log priority list

sabbaka opened the pull request:
I'm starting integration of session recording to Cockpit as it was agreed on meetings.
This is just initial step to changing priority list.
Approved design: https://github.com/cockpit-project/cockpit-design/blob/master/journal/journal-with-logins.png
Demo: https://youtu.be/u80ZfozGnDo

martinpitt commented on pkg/systemd/logs.html:
```diff
@@ -51,11 +51,15 @@
+          <li><a data-prio="0" translatable="yes">Emerg</a></li>
+          <li><a data-prio="1" translatable="yes">Alert</a></li>
+          <li><a data-prio="2" translatable="yes">Crit</a></li>
+          <li><a data-prio="3" translatable="yes">Err</a></li>
```
I'm not too happy about introducing these rather arbitrary abbreviations. There is no reason for that kind of brevity in the UI, and it's not clear how to translate them either. Maybe the explanations in `man syslog` would be a better basis for the strings here?

Could this keep "Everything" instead of "All" to avoid breaking all existing translations?

In the current version, repeating the severity levels and appending new ones is certainly bad too, and it's also not clear what the difference between "problem" and "error" is. I think something outside of the dropdown should point out that this applies to "this severity level or higher".

@andreasn,  @garrett, do you have some ideas here?

andreasn replied:
Patternfly has two important rules for abbreviations. One is to write only those that users are familiar with, and the other one is to use the consistently.
https://www.patternfly.org/styles/terminology-and-wording/#abbreviations

Our target audience of junior Linux admins and former Windows admins are not familiar with the Kernel log level abbreviations, so it's better to write them out in full.
It's also breaking the second rule, because "Warning" is written in full, but "Err" and "Crit" is not, so that's inconsistent.

sabbaka replied:
Thanks for feedback! I'll provide required changes.

sabbaka commented:
Updated. 
"Everything" is kept instead of "All".
Other error type names are written full now.

martinpitt commented:
The test/avocado/selenium-base.py needs to be adjusted:
```
08:30:27 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:596
08:30:27 ERROR| Traceback (most recent call last):
08:30:27 ERROR|   File "/tmp/avocado_tests/selenium-base.py", line 66, in test50ChangeTabLogs
08:30:27 ERROR|     "//span[@id='journal-prio' and contains(text(), '%s')]" % "Problems, Errors"))
08:30:27 ERROR|   File "/tmp/avocado_tests/seleniumlib.py", line 223, in wait_xpath
08:30:27 ERROR|     return self.wait(By.XPATH, text=el, baseelement=baseelement, overridetry=overridetry, fatal=fatal, cond=cond, jscheck=jscheck)
08:30:27 ERROR|   File "/tmp/avocado_tests/seleniumlib.py", line 212, in wait
08:30:27 ERROR|     raise Exception('ERR: Unable to locate name: %s' % str(text), screenshot_file)
08:30:27 ERROR| Exception: ("ERR: Unable to locate name: //span[@id='journal-prio' and contains(text(), 'Problems, Errors')]", 'screenshottest50ChangeTabLogs.png')
```

I'm still feeling a bit uneasy about dropping the "at least this level" aspect. I. e. before it was clear that e. g "Problems, Errors, Warnings" included different priorities down to the selected one, now it looks more like "show logs with exactly this level". @andreasn, WDYT?

Is this blocking your session recording work, or is it just an independent change?

andreasn commented on pkg/systemd/logs.html:
```diff
@@ -52,10 +52,14 @@
-          <li><a data-prio="4" translatable="yes">Problems, Errors, Warnings</a></li>
-          <li><a data-prio="3" translatable="yes">Problems, Errors</a></li>
-          <li><a data-prio="2" translatable="yes">Only Problems</a></li>
+          <li><a data-prio="0" translatable="yes">Emergency</a></li>
```
I would suggest "Only Emergency" here, to indicate the filtering out of all other messages.

martinpitt commented:
After speaking with @andreasn on IRC it seems better to turn the appropriate strings into e. g. "Warning and above; all but "Everything" and "Emergency".

andreasn commented on pkg/systemd/logs.html:
```diff
@@ -52,10 +52,14 @@
-          <li><a data-prio="3" translatable="yes">Problems, Errors</a></li>
-          <li><a data-prio="2" translatable="yes">Only Problems</a></li>
+          <li><a data-prio="0" translatable="yes">Emergency</a></li>
+          <li><a data-prio="1" translatable="yes">Alert</a></li>
```
It's a bit unclear if this will only show Alert, or Alerts and Emergency, so I would suggest this to be "Alert and above".
Same for Critical, Error, Warning, Notice, Info and Debug.

sabbaka commented:
I'll fix test and update text signs.
It's not that much of blocker, it's just a part of a new design of Logs page and as it's a simple one I decided to start with it.

sabbaka commented:
I've updated tests. Also, I have a question - should I also improve a bit those test to test all levels of logs - not only as it was "notice" at the end ( I've changed this to "debug" )?

martinpitt commented:
Thanks, this looks better. Running tests.

sabbaka commented:
Thanks!

martinpitt commented:
Screenshot for release notes:
![loglevels](https://user-images.githubusercontent.com/200109/37702158-ffbcf874-2cf1-11e8-8470-9e6c8eb70b81.png)

spbnick commented on pkg/systemd/logs.html:
```diff
@@ -52,10 +52,14 @@
-          <li><a data-prio="4" translatable="yes">Problems, Errors, Warnings</a></li>
-          <li><a data-prio="3" translatable="yes">Problems, Errors</a></li>
-          <li><a data-prio="2" translatable="yes">Only Problems</a></li>
+          <li><a data-prio="0" translatable="yes">Only Emergency</a></li>
```
Wouldn't it be awesome to have this say "Emergency Only" instead? Then we would have all items starting with the logging level name :)

spbnick commented on test/avocado/selenium-base.py:
```diff
@@ -63,42 +63,73 @@ def test50ChangeTabLogs(self):
         self.wait_id("journal-prio")
         self.click(self.wait_xpath(
-            "//span[@id='journal-prio' and contains(text(), '%s')]" % "Problems, Errors"))
+            "//span[@id='journal-prio' and contains(text(), '%s')]" % "Error and above"))
```
Hmm, how does that work with `translatable="yes"` in the HTML? I mean wouldn't that `"Error and above"` text be replaced with the translation, breaking this match?

martinpitt replied:
Tests run under C.utf8 (except for tests like check-pages which change the language explicitly), so that should be fine.
